zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterEventClient.java

```diff
@@ -86,6 +88,17 @@ public synchronized ResourceSet getAllResources() {
     }
   }
 
+  public synchronized List<ParagraphInfo> getParagraphList(AuthenticationInfo authInfo, String noteId) {
+    try {
+      String authInfoJson = gson.toJson(authInfo);
+      List<ParagraphInfo> paragraphList = intpEventServiceClient.getParagraphList(authInfoJson, noteId);
```

Comment:
Does intpEventServiceClient.getParagraphList throw exception when the current user doesn't have permission to read this note ?

Reply:
This should only catch the Thrift call exception.
If the user does not have permission to get this note, the function returns `null`.

Reply:
I think we should throw exception to tell user expclitly that he doesn't have permission on this note. Otherwise it would be very difficult to know why user get null.